IdentityCore/src/validation/MSIDAuthority.m

```diff
@@ -55,6 +99,22 @@ + (BOOL)isADFSInstanceURL:(NSURL *)endpointUrl
     return NO;
 }
 
++ (BOOL)isB2CInstanceURL:(NSURL *)endpointUrl
```

Comment:
in future we might want to let developer tell us, which authority they're giving us:
e.g. B2C, ADFS, AAD, etc
So, make sure this class is extensible  to support setting authority type from outside vs parsing the URL and looking for a "magic string" in it.

Reply:
as we discussed offline, we will do it in separate pr.